### PR TITLE
Fix issues with integration test mocking by extracting one layer of mocks

### PR DESCRIPTION
**Pull request by jkwatson-verta**
and cleaning up tests to use the new structure.

Note: there are still 5 legitimate, non-mockito-related test failures that I didn't try to address.

## Testing
- [ ] Unit test
- [ ] Deployed to dev env
- [ ] Other (explain) 

## Reverting
- [ ] Contains Migration - _Do Not Revert_

### CODE REVIEW

File: backend/server/src/test/java/ai/verta/modeldb/ModeldbTestConfigurationBeans.java [hunk 1]
```diff
@@ -132,4 +155,85 @@ private static TestConfig initializeTestConfig() {
     System.getProperties().put("server.port", testConfig.getSpringServer().getPort());
     return testConfig;
   }
+
+  private static class TestAuthServiceChannel extends AuthServiceChannel {
```
Comment (jkwatson-verta) [inline backend/server/src/test/java/ai/verta/modeldb/ModeldbTestConfigurationBeans.java:159]: this was the key to fixing all the breakages...basically removing one layer of mocking, so we're not mock->mock->mock any more

**Comment by anandjakhaniya** [comment]
> and cleaning up tests to use the new structure.
> 
> Note: there are still 5 legitimate, non-mockito-related test failures that I didn't try to address.
> 
> ## Impact and Context
> ## Risks and Area of Effect
> ## Testing
> * [ ]  Unit test
> * [ ]  Deployed to dev env
> * [ ]  Other (explain)
> 
> ## Reverting
> * [ ]  Contains Migration - _Do Not Revert_

it might be failed because we have enabled `permission_v2` flag and your PR Base branch does not fix the other test classes for the same excluding ProjectTest.

**Comment by anandjakhaniya** [comment]
@jkwatson-verta I have merged all PRs (https://github.com/VertaAI/modeldb/pull/3483) in one place with these PR changes and it seems it works but while testing at my local system, I see some tests are still behaving with flakiness.

**Review by evgeniilezhnin — APPROVED** [review]

**Comment by jkwatson-verta** [comment]
> @jkwatson-verta I have merged all PRs (#3483) in one place with these PR changes and it seems it works but while testing at my local system, I see some tests are still behaving with flakiness.

Can you be more specific? Can you point me at a branch with test flakiness?

**Comment by anandjakhaniya** [comment]
> > @jkwatson-verta I have merged all PRs (#3483) in one place with these PR changes and it seems it works but while testing at my local system, I see some tests are still behaving with flakiness.
> 
> Can you be more specific? Can you point me at a branch with test flakiness?

there is no specific test that always flakes but it randomly happens with any test Or no fails, if I am running Pull suites multiple times. sample failing is added in the attached screenshot which failed in this PR. 
image

**Comment by anandjakhaniya** [comment]
In my PR (https://github.com/VertaAI/modeldb/pull/3483) after merging all permission v2 stuff. Still, I am seeing flakiness. I have attached a screenshot.
image

**Comment by jkwatson-verta** [comment]
I poked at the deleteResources failures a bunch, but didn't find any great solution (or even an understanding of why it's happening). If we use ` -Dsurefire.rerunFailingTestsCount=2` from the command line, the test flakes seem to almost never happen twice in a row, so that might be our best bet for now.

**Comment by jkwatson-verta** [comment]
removing the mock for `deleteResources` on line 480 of ModelDbTestSetup did seem to make things slightly better.

**Comment by anandjakhaniya** [comment]
> removing the mock for `deleteResources` on line 480 of ModelDbTestSetup did seem to make things slightly better.

So should we update that in this PR?

**Comment by jkwatson-verta** [comment]
> > removing the mock for `deleteResources` on line 480 of ModelDbTestSetup did seem to make things slightly better.
> 
> So should we update that in this PR?

Probably not..just merge https://github.com/VertaAI/modeldb/pull/3493 into your branch

**Comment by anandjakhaniya** [comment]
@jkwatson-verta Could we merge this so I can update all other PRs and merge it? it is blocked since the last week. do you think we have to add something to this, still?

**Review by anandjakhaniya — APPROVED** [review]

**Pull request merged**